NLP/main/extract_display_quotes.py

```diff
@@ -0,0 +1,531 @@
+            )
+        
+        # widgets to select what to preview, i.e., speaker and/or quote and/or named entities
+        entity_options = widgets.HTML(
+            value="<b>Select which entity to show:</b>",
+            placeholder='',
+            description='',
+            )
+        
+        speaker_box = widgets.Checkbox(
+            value=False,
+            description='Speaker',
+            disabled=False,
+            indent=False,
+            layout=Layout(margin='0px 0px 0px 0px')
+            )
+        
+        quote_box = widgets.Checkbox(
+            value=False,
+            description='Quote',
+            disabled=False,
+            indent=False,
+            layout=Layout(margin='0px 0px 0px 0px')
+            )
+        
+        ne_box = widgets.Checkbox(
+            value=False,
+            description='Named Entities',
+            disabled=False,
+            indent=False,
+            layout=Layout(margin='0px 0px 0px 0px')
+            )
+        
+        # widget to show the preview
+        preview_button = widgets.Button(description='Click to preview', 
+                                        layout=Layout(margin='10px 0px 0px 10px'),
+                                        style=dict(font_style='italic',
+                                                   font_weight='bold'))
+        preview_out = widgets.Output()
+        
+        def on_preview_button_clicked(_):
+            with top_out:
+                clear_output()
+            with preview_out:                                                                                   
+                # what happens when we click the preview_button
+                clear_output()
+                text_id = text.value
+                show_what = []
+                if speaker_box.value:
+                    show_what.append('SPEAKER')
+                if quote_box.value:
+                    show_what.append('QUOTE')
+                if ne_box.value:
+                    show_what.append('NAMED ENTITIES')
+                if show_what==[]:
+                    print('Please select the entities to display!')
+                else:
+                    try:
+                        self.show_quotes(text_df, quotes_df, text_id, show_what, inc_ent, save_to_html=False)
+                    except:
+                        print('Please enter the correct text_id!')
+        
+        # link the preview_button with the function
+        show_what = preview_button.on_click(on_preview_button_clicked)
+        
+        # widget to save the preview
+        save_button = widgets.Button(description='Save preview', 
+                                     layout=Layout(margin='10px 0px 0px 10px'),
+                                     style=dict(font_style='italic',
+                                                font_weight='bold'))
+        
+        def on_save_button_clicked(_):
+            with preview_out:
```

Comment:
Is it possible to just save the current view without rerun the rendering? Also, I haven't tried, but if the notebook is running in binder, will the user be able to save/download the html file to local machine?

Reply:
- code updated to avoid re-run of rendering when saving the current view
- user in binder will be able to save & download to their local machine